addon/-private/system/store/record-data-wrapper.js

```diff
@@ -54,44 +65,68 @@ export default class RecordDataWrapper {
     return this.relationshipsDefinitionFor(modelName)[key]._inverseKey(this.store, modelClass);
   }
 
-  // TODO Igor David cleanup
   inverseIsAsyncForRelationship(modelName, key) {
     let modelClass = this.store.modelFor(modelName);
     return this.relationshipsDefinitionFor(modelName)[key]._inverseIsAsync(this.store, modelClass);
   }
 
-  notifyPropertyChange(modelName, id, clientId, key) {
-    let internalModel = this.store._getInternalModelForId(modelName, id, clientId);
+  // TODO IDENTIFIER RFC - arg should be identifier
+  notifyPropertyChange(type, id, lid, key) {
```

Comment:
Not sure if it's worth the churn and the temp object creation

Reply:
Follow Up:
---

_Today it:_

* Reduces the complexity of method signatures in the rest of the internals
* Ensures we have normalized `type` `id` `lid`
* Ensures that `type` `id` and `lid` are a valid combo

The following things are valid regardless of whether we make use of the _stable reference_ or move to a less constrained system. Since this is entirely non-user exposed though this shouldn't matter at the moment, and switching storage strategies with the `recordIdentifierFor` API is trivial.

_Tomorrow it:_

* Allows us to RFC `RecordDataWrapper` APIs that accept identifiers instead, which is closer to the `json-api` that `RecordData` users will be working with, especially for relationships.
* Allows us to reduce cognitive overhead by reducing method signatures (`method(identifier, propertyName) {}` is much easier than `method(type, id, lid, propertyName)`)